app/pages/BeelieversAuction/MintInfo.tsx

```diff
@@ -130,9 +131,23 @@ function MintAction({ isWinner, doRefund, hasMinted, setNftId, kiosk, setKiosk }
 		} catch (error) {
 			console.error("Error minting:", error);
 
+			const errorMessage = (error as Error).message;
+			let userMessage = "An error occurred during minting";
+
+			if (errorMessage) {
+				const parsedError = parseTxError(errorMessage);
+				if (parsedError && typeof parsedError === "object") {
+					userMessage = formatSuiMintErr(parsedError);
+				} else if (typeof parsedError === "string") {
+					userMessage = parsedError;
+				} else {
+					userMessage = formatSuiMintErr(error);
+				}
+			}
```

Comment:
let's move it to a function outisde of the component (will be easier to test and reduce code in the component)